C++/include/algorithm/string/heaps_algorithm.hpp

```diff
@@ -0,0 +1,49 @@
+/*
+    Heap's Algorithm
+    ----------------
+
+    This algorithm generates all possible
+permutations of a collection of n elements.    
+The algorithm minimizes movement: it generates each permutation from the previous one by interchanging a single pair of elements, while the other n-2 remain undisturbed.
+    
+    Time Complexity
+    ---------------
+    O(n!), where n is the number of elements in the collection
+
+    Space Complexity
+    ----------------
+    O(n), where n is the number of elements in the collection
+
+*/
+
+
+#ifndef HEAPS_ALGORITHM_HPP
+#define HEAPS_ALGORITHM_HPP
+
+#include <vector>
+
+using namespace std;
```

Comment:
Instead of `using namespace std` (which pulls in entire `std` namespace), please pull in only the identifiers you need. i.e. 

```
using std::string;
using std::vector;
```

You can also just prepend the identifier with `std::` whenever you use it. As in, `std::swap(collection[0],`.

Reply:
More info on this here: https://stackoverflow.com/questions/1452721/why-is-using-namespace-std-considered-bad-practice

Reply:
Thanks! I chose `using std::` for the sake of clean code.